packages/engine/src/physics/systems/PhysicsSystem.ts

```diff
@@ -134,8 +156,13 @@ export default async function PhysicsSystem(world: World) {
 
     // step physics world
     const substeps = engineState.physicsSubsteps.value
-    world.physicsWorld.timestep = engineState.fixedDeltaSeconds.value / substeps
+    const timestep = engineState.fixedDeltaSeconds.value / substeps
+    world.physicsWorld.timestep = timestep
+    const smoothAlpha = 50 * timestep
     for (let i = 0; i < substeps; i++) {
+      // smooth kinematic pose changes
+      for (const entity of kinematicPositionBodyQuery()) smoothKinematicBody(entity, smoothAlpha)
```

Comment:
Let's store the result of this query outside the substep loop to avoid multiple calls to bitecs.

Reply:
Only gets called once